.github/workflows/go.yml

```diff
@@ -12,40 +12,36 @@ jobs:
     steps:
 
     - name: Set up Go
-      uses: actions/setup-go@v2
+      uses: actions/setup-go@v5
       with:
         stable: 'false'
         go-version: '1.21'
 
     - name: Checkout
-      uses: actions/checkout@v1
+      uses: actions/checkout@v4
       with:
         fetch-depth: 1
 
     - name: Cache Go modules
-      uses: actions/cache@v1
+      uses: actions/cache@v4
       with:
-        path: ~/go/pkg/mod
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
         key: ${{ runner.os }}-build-${{ hashFiles('**/go.sum') }}
         restore-keys: |
-          ${{ runner.OS }}-build-${{ env.cache-name }}-
           ${{ runner.OS }}-build-
```

Comment:
Saw in https://github.com/janos/singleflight/actions/runs/7099504482/job/19323630907 that the caching of go modules doesn't seem to be working properly